### DISCUSSION TITLE
Future Post Dates on Events do not effect anything

### DISCUSSION THREAD

**Discussion by andypullen** [solspace/craft-calendar#339] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
Setting the Post Date of an Event to a future date does not change the status of the Event like other Craft entries do. Additionally, the `postDate` is not accessible in twig templates, so building queries to exclude Events with future Post Dates is also not possible.

**Steps to reproduce**

1. Create an Event with a future Post Date
2. Note that the Event status does not change to Pending, but is Live instead

**Expected behavior**
- Putting a future date in the Post Date field should change the Event status to Pending
- Pending Events should not be included in frontend Event queries by default
- The `postDate` value should be included in the Event object in twig

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.12.3
- Calendar Version: 5.0.8
- Calendar Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment, marked as answer]
Sorry for the delay @andypullen,

Calendar Events are handled a little bit differently from Craft Entries. However, you can work around this by doing something like this in your template:

``` twig
{% set ignoreFutureDates = date('today')|atom %}

<h3>List of Events</h3>
{% set eventOptions = {
    rangeStart: targetStartDate,
    rangeEnd: targetEndDate,
    postDate: '<' ~ ignoreFutureDates,
    limit: 15
} %}
```

I tested this and it seems to work well for me. 🙂

**Comment by andypullen** [comment]
I will give this a shot — thanks for the tip and help!
